dvc/project.py

```diff
@@ -165,10 +165,22 @@ def install(self):
     def to_dvc_path(self, path):
         return os.path.relpath(path, self.root_dir)
 
-    def _check_output_duplication(self, outs):
+    def _check_cwd_specified_as_output(self, cwd, stages=None):
```

Comment:
We can safely make `stages` a required argument, instead of it being optional and not collect it here(and in the _check_output_duplication) with `stages = stages or self.stages()`, since there functions are currently only used with stages passed to them.